.changeset/quick-dragons-deliver.md

```diff
@@ -0,0 +1,76 @@
+---
+'merchant-center-application-template-starter': minor
+'@commercetools-frontend/application-config': minor
+'@commercetools-frontend/application-shell': minor
+'@commercetools-frontend/constants': minor
+'playground': minor
+---
+
+Allow to pass the application routes as `children` of `<ApplicationShell>`, instead of using the `render` prop.
+When doing so, the application entry point routes are automatically configured according to the `entryPointUriPath` defined in the `custom-application-config.json`.
+
+> Note that it's still possible to use the `render` prop. However, for most of the cases it should suffice to rely on the pre-configured routes.
+
+```diff
+const AsyncApplicationRoutes = React.lazy(
+  () => import('../../routes' /* webpackChunkName: "starter-routes" */)
+);
+
+-export const ApplicationStarter = () => (
+-  <Switch>
+-    {
+-      /* For development, it's useful to redirect to the actual
+-      application routes when you open the browser at http://localhost:3001 */
+-      process.env.NODE_ENV === 'production' ? null : (
+-        <Redirect
+-          exact={true}
+-          from="/:projectKey"
+-          to="/:projectKey/examples-starter"
+-        />
+-      )
+-    }
+-    <Route
+-      path="/:projectKey/examples-starter"
+-      component={AsyncApplicationRoutes}
+-    />
+-    {/* Catch-all route */}
+-    <RouteCatchAll />
+-  </Switch>
+-);
+-ApplicationStarter.displayName = 'ApplicationStarter';
+
+const EntryPoint = () => (
+  <ApplicationShell
+    environment={window.app}
+    onRegisterErrorListeners={({ dispatch }) => {
+      Sdk.Get.errorHandler = (error) =>
+        globalActions.handleActionError(error, 'sdk')(dispatch);
+    }}
+    applicationMessages={loadMessages}
+    DEV_ONLY__loadNavbarMenuConfig={() =>
+      import('../../../menu.json').then((data) => data.default || data)
+    }
+    featureFlags={FEATURE_FLAGS}
+-    render={() => <ApplicationStarter />}
+-  />
++  >
++    <AsyncApplicationRoutes />
++  </ApplicationShell>
+);
+```
+
+Furthermore, the `test-utils` of the `@commercetools-frontend/application-shell` have now a new option to enable this opt-in behavior of rendering the application with pre-configured routes.
+To enable this option, pass the `renderEntryPointRoutes: true` to the `renderApp` or `renderAppWithRedux` functions.
```

Comment:
Nitty: `renderEntryPointRoutes` sounds a bit like a render-prop to me while it obviously is a boolean option. I assume we would like to enable this behaviour by default at some point. I personally (subjective obviously) started to like `disableX` for such things. So this would be defaulting to `disableAutomaticEntryPointRoutes` which defaults to `true`. One can then opt-into the new behaviour while we with some breaking release only claim that the option was now removed and is always is true.

Reply:
Yeah I couldn't come up with a better name. Thanks for the suggestion, I like it!

Reply:
I changed it as per suggestion.